### ISSUE TITLE
Make the learning rate constant configurable in NoamTrainer

### ISSUE THREAD

**Issue by DevSinghSachan** [neulab/xnmt#471]
https://github.com/neulab/xnmt/blob/1324881a5a5db5ab6bf3a05376d62592cce70fd7/xnmt/optimizer.py#L241

self.optimizer.learning_rate = 1. * decay

Let's the constant 1. in the above equation as a configuration option. It's name can be 'learning_rate_constant' with its default value 2.0

**Comment by DevSinghSachan** [comment]
Also, let's change the default values of beta_2 to 0.997 and of warmup steps to 16000, as they are the default values in tensor2tensor for single GPU case.

**Comment by neubig** [comment]
@DevSinghSachan Welcome a PR for this.
